sources/sqlserver/README.md

```diff
@@ -70,3 +70,86 @@ for SQLserver source. Password can be provided at the password prompt.
+primary keys for all tables, but does not enforce this. When converting a table
+without a primary key, HarbourBridge will create a new primary key of type
+INT64. By default, the name of the new column is `synth_id`. If there is already
+a column with that name, then a variation is used to avoid collisions.
+Alternatively, when converting a table without a Primary Key having a `UNIQUE`
+constraint, HarbourBridge will automatically choose that column as a Primary Key
+column instead of generating a new one. In case of multiple unique constraints,
+it can pick any.
+
+### NOT NULL Constraints
+
+The tool preserves `NOT NULL` constraints. Note that Spanner does not require
+primary key columns to be `NOT NULL`. However, in SQL Server, a primary key is a
+combination of `NOT NULL` and `UNIQUE`, and so primary key columns from
+SQL Server will be mapped to Spanner columns that are both primary keys and `NOT NULL`.
+
+### Foreign Keys
+
+The tool maps SQL Server foreign key constraints into Spanner foreign key constraints, and
+preserves constraint names where possible. Since Spanner doesn't support `DELETE CASCADE`
+and `UPDATE CASCADE` actions, we drop them.
+
+### Default Values
+
+Spanner does not currently support default values. We drop these
+SQL Server features during conversion.
+
+### Secondary Indexes
+
+The tool maps SQL Server non-clustered indexes to Spanner secondary indexes, and preserves
+constraint names where possible. Note that Spanner requires index key constraint
+names to be globally unique (within a database), but in SQL Server they only have to be
+unique for a table, so we add a uniqueness suffix to a name if needed. The tool also
+maps `UNIQUE` constraint into `UNIQUE` secondary index.
+
+### Other SQL Server features
+
+SQL Server has many other features we haven't discussed, including functions,
+sequences, procedures, triggers and views which are currently not supported in Spanner. 
+The tool does not support these and the relevant schema info is ignored during schema
+conversion. 
+
+## Data Conversion
+
+### Strings, character set support and UTF-8
+
+Spanner requires that `STRING` values be UTF-8 encoded. All Spanner functions
+and operators that act on `STRING` values operate on Unicode characters rather
+than bytes. Since we map many SQL Server types (including `VARCHAR`)
+to Spanner's `STRING` type, HarbourBridge is effectively a UTF-8 based
+tool.
+Internally, we use Go's string type, which supports UTF-8.
+
+### Timestamps and Timezones
+
+There are some subtle differences in how timestamps are
+handled in SQL Server and Spanner.
+
+During data conversion, SQL Server `SMALLDATETIME`, `DATETIME2`, `DATETIME` values 
+in ISO 8601 format without offsets which is coherent with Spanenr. Hence, they are
+directly stored in Spanner. For `DATETIMEOFFSET`, the value contains an offset.
+During conversion, the timezone is respected during the conversion to UTC.
```

Comment:
Did you mean, "During conversion the timezone offset is taken into account and the corresponding UTC time is saved ?"